src/main/java/org/chocosolver/solver/search/strategy/assignments/DecisionOperator.java

```diff
@@ -63,175 +65,14 @@
-            var.instantiateTo(value, cause);
-        }
-
-        @Override
-        public void unapply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.removeValue(value, cause);
-        }
-
-        @Override
-        public String toString() {
-            return " == ";
-        }
-
-        @Override
-        public boolean isValid(IntVar var, int value) {
-            return var.contains(value);
-        }
-
-        @Override
-        public DecisionOperator opposite() {
-            return int_neq;
-        }
-    };
-
-    DecisionOperator<IntVar> int_neq = new DecisionOperator<IntVar>() {
-
-        @Override
-        public void apply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.removeValue(value, cause);
-        }
-
-        @Override
-        public void unapply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.instantiateTo(value, cause);
-        }
-
-        @Override
-        public String toString() {
-            return " != ";
-        }
-
-        @Override
-        public boolean isValid(IntVar var, int value) {
-            return var.contains(value);
-        }
-
-        @Override
-        public DecisionOperator opposite() {
-            return int_eq;
-        }
-    };
-
-    DecisionOperator<IntVar> int_split = new DecisionOperator<IntVar>() {
-
-        @Override
-        public void apply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.updateUpperBound(value, cause);
-        }
-
-        @Override
-        public void unapply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.updateLowerBound(value + 1, cause);
-        }
-
-        @Override
-        public String toString() {
-            return " <= ";
-        }
-
-        @Override
-        public boolean isValid(IntVar var, int value) {
-            return var.getUB() >= value;
-        }
-
-        @Override
-        public DecisionOperator opposite() {
-            return int_reverse_split;
-        }
-
-    };
-
-    DecisionOperator<IntVar> int_reverse_split = new DecisionOperator<IntVar>() {
-
-        @Override
-        public void apply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.updateLowerBound(value, cause);
-        }
-
-        @Override
-        public void unapply(IntVar var, int value, ICause cause) throws ContradictionException {
-            var.updateUpperBound(value - 1, cause);
-        }
-
-        @Override
-        public String toString() {
-            return " >= ";
-        }
-
-        @Override
-        public boolean isValid(IntVar var, int value) {
-            return var.getLB() <= value;
-        }
-
-        @Override
-        public DecisionOperator opposite() {
-            return int_split;
-        }
-    };
-
+    DecisionOperator<IntVar> int_eq = DecisionOperatorFactory.makeIntEq();
```

Comment:
We should consider calling directly the factory and remove these methods from DecisionOperator